### PR TITLE
feature(manager): multi-keyspace data validation with latte for manager sanity

### PR DESCRIPTION
**Pull request by mikliapko**
Closes https://scylladb.atlassian.net/browse/CLOUD-400

## Summary

- Add a multi-keyspace latte workload covering all CQL data types across 2 keyspaces × 2 tables (+ SI + MV) for deterministic backup/restore validation
- Unify `test_basic_backup` / `test_restore_backup_with_task` into a single `test_backup_and_restore(restore_with_task)` method; remove hardcoded `MULTI_REGION` branch from inside the method
- Auto-discover ks/table mapping in `verify_backup_success` via `get_non_system_ks_cf_list` instead of assuming `standard1`
- Skip Manager-task restore on multi-DC clusters in `test_manager_sanity` (nodetool refresh restore always runs)
- Add `manager-sanity-multidc-latte.yaml` test config and wire the ubuntu24 sanity jenkinsfile to use it

For now, we enable latte generated dataset for Ubuntu24 sanity only but later on will extend it across the whole Manager tests.

## Testing

- [x] [ubuntu24-sanity-test with latte](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/mikita/job/manager-master-clone/job/ubuntu24-sanity-test/15/)
- [x] [regression for vnodes + tablets test](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/mikita/job/manager-master-clone/job/ubuntu24-vnodes-tablets-enterprise-sanity-test/5/)

### CODE REVIEW

File: sdcm/mgmt/operations.py [hunk 1]
```diff
@@ -755,30 +755,43 @@ def verify_backup_success(
...
+        ks_tables_map: dict[str, list[str]] = {}
+
+        ks_cf_list = self.db_cluster.get_non_system_ks_cf_list(
+            db_node=self.db_cluster.nodes[0],
+            filter_out_mv=True,
+            filter_by_keyspace=ks_names,
+        )
+        for ks_cf in ks_cf_list:
+            ks, table = ks_cf.split(".")
+            ks_tables_map.setdefault(ks, []).append(table)
```
[the remark this answers is not in the document]
Comment (mikliapko) [reply sdcm/mgmt/operations.py:772]: make sense, will improve

File: sdcm/mgmt/operations.py [hunk 2]
```diff
@@ -755,30 +755,43 @@ def verify_backup_success(
...
+        self.log.debug(f"ks_tables_map: {ks_tables_map}")
```
[the remark this answers is not in the document]
Comment (mikliapko) [reply sdcm/mgmt/operations.py:773]: In practice, the keyspace/table names used in manager sanity tests are plain identifiers - `cql_quote_if_needed` returns them unchanged. This is a theoretical edge case, not a real risk for this PR.

File: sdcm/mgmt/operations.py [hunk 3]
```diff
@@ -755,30 +755,43 @@ def verify_backup_success(
...
+        snapshot_tag = backup_task.get_snapshot_tag()
+        ks_tables_map: dict[str, list[str]] = {}
+
+        ks_cf_list = self.db_cluster.get_non_system_ks_cf_list(
+            db_node=self.db_cluster.nodes[0],
+            filter_out_mv=True,
+            filter_by_keyspace=ks_names,
+        )
+        for ks_cf in ks_cf_list:
+            ks, table = ks_cf.split(".")
```
[the remark this answers is not in the document]
Comment (mikliapko) [reply sdcm/mgmt/operations.py]: fixed

File: sdcm/mgmt/operations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikliapko) [reply sdcm/mgmt/operations.py:772]: fixed

File: data_dir/latte/mgmt_data_validation.rn [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+                // Stored as string like "1mo1d1h1m1s", returned as
+                // {"days": N, "months": N, "nanoseconds": N}.
+                continue
+            } else if col_name == "col_date" {
+                // TODO: implement proper comparison for Date CQL type.
+                continue
+            } else if col_name == "col_time" {
+                // TODO: implement proper comparison for Time CQL type.
+                continue
+            }
```
[the remark this answers is not in the document]
Comment (mikliapko) [reply data_dir/latte/mgmt_data_validation.rn:472]: it's a known issue

File: mgmt_cli_test.py [hunk 5]
```diff
@@ -985,31 +983,39 @@ class ManagerSanityTests(
...
-        with self.subTest("Restore Backup Test"):
-            self.test_restore_backup_with_task(ks_names=ks_names)
-        with self.subTest("Repair Multiple Keyspace Types"):
+
+        if not is_multi_dc_cluster:
+            with self.subTest("Backup and restore (via nodetool refresh, out of Manager) test"):
+                self.test_backup_and_restore(restore_with_task=False, ks_names=ks_names)
+        with self.subTest("Backup and restore (via Manager task) test"):
+            self.test_backup_and_restore(restore_with_task=True, ks_names=ks_names)
+        with self.subTest("Repair multiple keyspace types test"):
```
[the remark this answers is not in the document]
Comment (mikliapko) [reply mgmt_cli_test.py:1006]: fixed docstring

File: data_dir/latte/mgmt_data_validation.rn [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+        kc=KEYSPACE_COUNT, tc=TABLE_COUNT_PER_KEYSPACE,
+        total=KEYSPACE_COUNT * TABLE_COUNT_PER_KEYSPACE,
+    );
+}
+
+// ========================================
+// erase(db) - truncate all tables
+// ========================================
+
+pub async fn erase(db, i) {
```
Comment (vponomaryov) [inline data_dir/latte/mgmt_data_validation.rn]: The `erase` fn name is reserved name to be called within `latte load` command.
And that command has only one parameter.

So, need to rename it to avoid confusions.

File: data_dir/latte/mgmt_data_validation.rn [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+/// Uses 'seed_offset' derived from (ks_idx, t_idx) to make data unique per table.
+/// The 'idx' controls null distribution (same pattern across tables),
+/// while 'data_seed' controls actual values (unique per table).
+async fn generate_row_data(db, row_i, ks_idx, t_idx) {
+    let idx = row_i % ROW_COUNT_PER_TABLE;
+    let partition = db.get_partition_info("main", idx).await;
+
+    // Per-table seed offset ensures unique data across tables (same approach as schema_scale.rn)
+    let seed_offset = hash2(ks_idx, t_idx) / 2;
+    let data_seed = idx + seed_offset;
```
Comment (vponomaryov) [inline data_dir/latte/mgmt_data_validation.rn]: It is too obfuscated for human prediction.
We know index of keyspace, index of table and number of rows in it.
So, use these directly.
For example, having 50k rows per table we can have following unique values per table:
- ks=1, t=1 -> offset=0
- ks=1, t=2 -> offset=50000
- ks=2, t=1 -> offset=100000
- ks=2, t=2 -> offset=150000

File: data_dir/latte/mgmt_data_validation.rn [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+        return
+    }
+    validate_row_columns(db, d, rows.0, err_info).await?;
+}
+
+/// Validate base table data (column-by-column) round-robin across all tables.
+/// SI and MV existence is verified at prepare() time — if either is missing after
+/// restore, prepare() fails hard before validation starts.
+/// Set -d to row_count_per_table * total_tables.
+pub async fn validate(db, i) {
```
Comment (vponomaryov) [inline data_dir/latte/mgmt_data_validation.rn:521]: Why duplicating the `get_by_ck` rune function?

File: data_dir/latte/mgmt_data_validation.rn (hunk 6 above)
Comment (mikliapko) [reply data_dir/latte/mgmt_data_validation.rn]: renamed to `truncate_all`

File: data_dir/latte/mgmt_data_validation.rn (hunk 7 above)
Comment (mikliapko) [reply data_dir/latte/mgmt_data_validation.rn]: fixed replacing by `let seed_offset = (ks_idx * TABLE_COUNT_PER_KEYSPACE + t_idx) * ROW_COUNT_PER_TABLE;`

File: data_dir/latte/mgmt_data_validation.rn (hunk 8 above)
Comment (mikliapko) [reply data_dir/latte/mgmt_data_validation.rn:521]: removed this function

**Comment by mikliapko** [comment]
@scylladb/qa-maintainers Could you please take a look?

**Review by vponomaryov — APPROVED** [review]
LGTM

**Review by pehala — APPROVED** [review]
